src/Sylius/Behat/Context/Ui/Admin/ManagingSlideshowBlocksContext.php

```diff
@@ -0,0 +1,233 @@
+ */
+final class ManagingSlideshowBlocksContext implements Context
+{
+    /**
+     * @var IndexPageInterface
+     */
+    private $indexPage;
+
+    /**
+     * @var CreatePageInterface
+     */
+    private $createPage;
+
+    /**
+     * @var UpdatePageInterface
+     */
+    private $updatePage;
+
+    /**
+     * @param IndexPageInterface $indexPage
+     * @param CreatePageInterface $createPage
+     * @param UpdatePageInterface $updatePage
+     */
+    public function __construct(
+        IndexPageInterface $indexPage,
+        CreatePageInterface $createPage,
+        UpdatePageInterface $updatePage
+    )
+    {
+        $this->indexPage = $indexPage;
+        $this->createPage = $createPage;
+        $this->updatePage = $updatePage;
+    }
+
+    /**
+     * @Given I want to create a new slideshow block
+     * @Given I want to add a new slideshow block
+     */
+    public function iWantToCreateNewSlideShow()
+    {
+        $this->createPage->open();
+    }
+
+    /**
+     * @When I want to browse slideshow blocks of the store
+     */
+    public function iWantToBrowseSlideShowsOfTheStore()
+    {
+        $this->indexPage->open();
+    }
+
+    /**
+     * @When I set its title to :title
+     */
+    public function iSetItsTitleTo($title)
+    {
+        $this->createPage->setTitle($title);
+    }
+
+    /**
+     * @When I set its name to :name
+     */
+    public function iSetItsNameTo($name)
+    {
+        $this->createPage->setName($name);
+    }
+
+    /**
+     * @When I make it published
+     */
+    public function iMakeItPublished()
+    {
+        $this->createPage->setPublished();
+    }
+
+    /**
+     * @When I make it available from :startsDate to :endsDate
+     */
+    public function iMakeItAvailableFromTo(\DateTime $startsDate, \DateTime $endsDate)
+    {
+        $this->createPage->setPublishStartDate($startsDate);
+        $this->createPage->setPublishEndDate($endsDate);
+    }
+
+    /**
+     * @When I add a slide available from :startsDate to :endsDate
+     */
+    public function iAddASlide(\DateTime $startsDate, \DateTime $endsDate)
+    {
+        $this->createPage->addSlide($startsDate, $endsDate);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @Then /^I should be notified that (name|title) is required$/
+     */
+    public function iShouldBeNotifiedThatElementIsRequired($element)
+    {
+        Assert::same(
+            $this->createPage->getValidationMessage($element),
+            'This value should not be blank.'
+        );
+    }
+
+    /**
+     * @Then the slideshow block :title should appear in the store
+     * @Then I should see the slideshow block :title in the list
+     */
+    public function theSlideShowShouldAppearInTheStore($title)
+    {
+        if (!$this->indexPage->isOpen()) {
+            $this->indexPage->open();
```

Comment:
If you expect that you will be on index page after successful create action maybe you should assert that. This may cuz false positive result. ``Assert::true($this->indexPage->isOpen())``. WDYT?